src/v/pandaproxy/client/client.cc

```diff
@@ -134,4 +138,86 @@ ss::future<kafka::produce_response::partition> client::produce_record_batch(
     return _producer.produce(std::move(tp), std::move(batch));
 }
 
+kafka::fetch_response::partition
+make_fetch_response(model::topic_partition tp, std::exception_ptr ex) {
+    kafka::error_code error;
+    try {
+        std::rethrow_exception(std::move(ex));
+    } catch (const partition_error& ex) {
+        vlog(ppclog.debug, "handling partition_error {}", ex.what());
+        error = ex.error;
+    } catch (const broker_error& ex) {
+        vlog(ppclog.debug, "handling broker_error {}", ex.what());
+        error = ex.error;
+    } catch (const ss::gate_closed_exception&) {
+        vlog(ppclog.debug, "gate_closed_exception");
+        error = kafka::error_code::operation_not_attempted;
```

Comment:
does this correspond to a shutdown scenario? i'm curious why this particular kafka error message was chosen. i don't have an alternative suggestion, this is just the first time i've actually seen this error code used.

Reply:
Yeah. It might not be right, but I couldn't find anything more suitable.

```
The broker did not attempt to execute this operation. This may happen for batched RPCs where some operations in the batch failed, causing the broker to respond without trying the rest
```
Semantically it felt right, and a bit nicer than just dropping the connection.